### PR TITLE
Add scenarios for Quarkus CLI

### PR DESCRIPTION
**Pull request by Sgitario**
The first set of scenarios include:
- Check Quarkus CLI version
- Create / Build applications with/out extensions. On JVM and Native
- Run applications on DEV mode
- Check completion installation
- Check Help option for some commands
- Check with special characters
- Check list of extensions when not in a current Quarkus project

### CODE REVIEW

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliHelpIT.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+public class QuarkusCliHelpIT {
+
+    static final String HELP_COMMAND = "--help";
+
+    @Inject
+    static QuarkusCliClient cliClient;
+
+    @Test
+    public void shouldCommandsHaveHelpOption() {
+        for (CommandWithHelp command : CommandWithHelp.values()) {
```
Comment (Sgitario) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliHelpIT.java:26]: Another approach could be using a `@ParametrizedTest` and `@ArgumentsSource`.

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliSpecialCharsIT.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            "Îñţérñåţîöñåļîžåţîờñ");
+
+    @Inject
+    static QuarkusCliClient cliClient;
+
+    private QuarkusCliClient.Result result;
+
+    @Test
+    public void shouldCreateApplicationOnJvmWithSpaces() {
+        for (String folder : SPECIAL_CHARS_FOLDERS) {
```
Comment (Sgitario) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliSpecialCharsIT.java:48]: Another approach could be using a `@ParametrizedTest` and `@ArgumentsSource`.

**Comment by rsvoboda** [comment]
Running `mvn clean verify -pl .,quarkus-cli -Dinclude.quarkus-cli-tests -Dquarkus.qe.framework.version=0.0.2`, using 2.0.0.CR3 based `quarkus` command

Getting:
```
[ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingConcise:55->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
[ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingDefaults:43->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
[ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingFull:61->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
[ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingOrigins:49->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
[ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingOtherPlatformVersions:68->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Unknown option: '--platform-bom=io.quarkus:quarkus-bom:1.13.4.Final' ==> expected: <true> but was: <false>
[ERROR]   QuarkusCliVersionIT.shouldVersionMatchQuarkusVersion:39 expected: <Client Version 999-SNAPSHOT> but was: <Client Version 2.0.0.CR3>
```

Is `whenGetListExtensions` specific to cli version 999-SNAPSHOT ?

**Comment by Sgitario** [comment]
> Running `mvn clean verify -pl .,quarkus-cli -Dinclude.quarkus-cli-tests -Dquarkus.qe.framework.version=0.0.2`, using 2.0.0.CR3 based `quarkus` command
> 
> Getting:
> 
> ```
> [ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingConcise:55->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
> [ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingDefaults:43->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
> [ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingFull:61->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
> [ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingOrigins:49->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Is this a project directory? Unable to find a build file in: /Users/rsvoboda/tmp/quarkus-test-suite/quarkus-cli/target ==> expected: <true> but was: <false>
> [ERROR]   QuarkusCliExtensionsIT.shouldListExtensionsUsingOtherPlatformVersions:68->whenGetListExtensions:92->assertResultIsSuccessful:125 Extensions list command didn't work. Output: [ERROR] ❗ Unable to list extensions: Unknown option: '--platform-bom=io.quarkus:quarkus-bom:1.13.4.Final' ==> expected: <true> but was: <false>
> [ERROR]   QuarkusCliVersionIT.shouldVersionMatchQuarkusVersion:39 expected: <Client Version 999-SNAPSHOT> but was: <Client Version 2.0.0.CR3>
> ```
> 
> Is `whenGetListExtensions` specific to cli version 999-SNAPSHOT ?

Yes, these changes are not released yet.

**Comment by rsvoboda** [comment]
One more thing, do you know if there will be native variant of cli distributed or just java one?

**Comment by Sgitario** [comment]
> One more thing, do you know if there will be native variant of cli distributed or just java one?

For now, just jar / jbang. In the future, more options are coming: https://github.com/quarkusio/quarkus/issues/10354

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliSpecialCharsIT.java (hunk 2 above)
Comment (rsvoboda) [reply quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliSpecialCharsIT.java:48]: I prefer individual test for the string like in https://github.com/quarkus-qe/quarkus-startstop/blob/main/testsuite/src/it/java/io/quarkus/ts/startstop/SpecialCharsTest.java#L159

Comment (Sgitario) [reply quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliSpecialCharsIT.java:48]: Changed

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**